tests/lib/test_openstack_image.py

```diff
@@ -349,3 +351,95 @@ def __getitem__(self, item):
         result = self.instance.find_non_existent_projects(cloud_account="test")
 
         self.assertEqual(result, {"ProjectID1": ["ImageID1", "ImageID2"]})
+
+    @raises(ValueError)
+    def test_email_users_no_email_error(self):
```

Comment:
...

Splitting this into a parent class would also allow us to remove duplication of these tests too